tests/e2e/cht/pregnancy.spec.js

```diff
@@ -0,0 +1,136 @@
+const utils = require('../../utils');
+const commonElements = require('../../page-objects/common/common.po');
+const loginPage = require('../../page-objects/login/login.po');
+const contactsPage = require('../../page-objects/contacts/contacts.po');
+const helper = require('../../helper');
+const analyticsPo = require('../../page-objects/analytics/analytics.po');
+const moment = require('moment');
+const Factory = require('rosie').Factory;
+const formFiller = require('../../form-filling').fillForm;
+require('../../factories/cht/reports/pregnancy.js');
+require('../../factories/cht/reports/pregnancy-visit');
+require('../../factories/cht/reports/delivery');
+require('../../factories/cht/users/users');
+require('../../factories/cht/contacts/person');
+const place = require('../../factories/cht/contacts/place');
+const dateFormat = 'D MMM, YYYY';
+
+const genericFormPo = require('../../page-objects/forms/generic-form.po');
+
+
+const places = place.generateHierarchy();
+const healthCenter = places.find((place) => place.type === 'health_center');
+const clinic = places.find((place) => place.type === 'clinic');
+const pregnancyWoman = Factory.build('person',
+  {
+    parent: {
+      _id: clinic._id,
+      parent: clinic.parent
+    }
+  });
+
+const offlineUser = Factory.build('offlineUser', {
+  place: healthCenter._id
+});
+
+
+const docs = [...places, pregnancyWoman];
+
+
+describe('Pregnancy workflow on cht : ', () => {
+  let originalTimeout;
+  beforeEach(function() {
+    originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 2 * 60 * 1000;
+  });
+
+  afterEach(function() {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+  });
+
+  beforeAll(async () => {
+    await utils.saveDocs(docs);
+    await utils.createUsers([offlineUser]);
+  });
+
+  it('should register a pregnancy', async () => {
+    await commonElements.goToLoginPageNative();
+    await loginPage.loginNative(offlineUser.username, offlineUser.password);
+    await utils.closeTour();
+    await commonElements.goToPeople();
+    await contactsPage.selectLHSRowByText(clinic.name);
+    await contactsPage.selectContactByName(pregnancyWoman.name);
+    await helper.clickElementNative(contactsPage.newActions);
+    await helper.clickElementNative(contactsPage.formById('pregnancy'));
+    await helper.waitUntilReadyNative(genericFormPo.formTitle);
+    const pregnancyReport = Factory.build('pregnancy');
```

Comment:
I'm not a big fan of magic strings here - ie: if you mistyped pergnancy then you'd presumably get a runtime error. Instead of calling `Factory.build` directly can we provide a named function which does the same, eg:

```
module.exports = {
  buildPregnancy: opts => Factory.build('pregnancy', opts);
};
```

That way you get code completion from your IDE too!

Reply:
@garethbowen for such an approach, Rosie has a [direct way to do "unregistered" factories](https://github.com/rosiejs/rosie#usage-in-nodejs). Might that be preferable?

Reply:
YES! Perfect.